recipes/yaggo/meta.yaml

```diff
@@ -14,8 +14,8 @@ build:
 
 requirements:
   run:
-    - ruby
-    - jemalloc
+    # artificial version constraint to avoid the outdated bioconda::ruby
+    - ruby >2.2.3
```

Comment:
This is a ugly workaround but should be fine, IMO.
I'll work on a better solution either in Bioconda or in `conda` soon, I guess.

Reply:
This is what we recommend, if we can not remove the package from our channel. Thanks @mbargull!

Reply:
At some point we should discuss whether we want to change the channel priority from `[bioconda, conda-forge, defaults]` to `[conda-forge, bioconda, defaults]`. That would solve quite a few issues (including "solver getting stuck" ones). It of course has implications if we had package name conflicts with `conda-forge` (but thankfully we lint for that on `bioconda` as well as `conda-forge`).
The other possibility would be for me to implement (or persuade someone else to do so :wink:) some kind of selective channel priority, i.e., being able to specify `[[bioconda, conda-forge], defaults]` as a channel priority list (of lists) that expresses that `bioconda` and `conda-forge` have the same prio. Meaning like `--no-channel-priority`, but selectively only for `bioconda` and `conda-forge`.

Reply:
I'm all in for the first one, but you need to discuss this with @johanneskoester, he had objections last time against this :)
The second idea is problematic, as during install time you need to know this, which is not ideal for our users imho.
We could also just mark certain packages as deprecated ...

Reply:
Yeah, I'd definitely want some more opinions from the core team on this -- because I can get behind the reservations on this one. And it's furthermore a "hard to deploy" change since we are also talking about telling the user to yet again change their config...

Though, I'm not yet seeing how that "selective equal channel priority" thing is more problematic than telling the user to use `[bioconda, conda-forge, defaults]` or `[conda-forge, bioconda, defaults]`. All of those are a one-time configuration thing (or memorizing two CLI flags if they don't use a config).

Reply:
Ah your second comment was not about a per recipe change. Mh, have you tried equal priorities? What happens in the case of openjdk than? This sounds dangerous random to me :)

Reply:
> What happens in the case of openjdk than? This sounds dangerous random to me :)

Exactly because of that I wouldn't want to use `--no-channel-priority` :wink:. I was thinking about being able to set equal priorities for a *subset* of channels, i.e., `bioconda` and `conda-forge` only. This functionality is not yet available in `conda`, but I have some different use cases in mind (e.g., `(conda-canary, defaults)`) where it would make sense to have it.
Currently `conda`'s solver (roughly) sets priority via
```python
(channel_prio, version, build_number, ...)
```
or if you provide `--no-channel-priority`, it's
```python
(version, channel_prio, build_number, ...)
```
. What I want is something like
```python
(main_channel_prio, version, equi_channel_prio, build_number, ...)
```
with, e.g.,
```python
main_channel_prio = {('bioconda', 'conda-forge'): 0, ('defaults',): 1}
equi_channel_prio[('bioconda', 'conda-forge')] = {'bioconda':0, 'conda-forge': 1}
```
if that makes sense to you.
Deep down in the solver's code it should be fairly simple to implement. But I'm not sure what other parts of `conda`'s code are affected by this (e.g., `subdir`s being treated as subchannels or something). I only had the idea a while back but didn't look into it yet. I'll have to ask @kalefranz some time, I think.